larq_compute_engine/mlir/transforms/prepare_patterns_common.td

```diff
@@ -4,9 +4,18 @@ include "tensorflow/compiler/mlir/tensorflow/ir/tf_ops.td"
 include "larq_compute_engine/mlir/ir/lce_ops.td"
 include "larq_compute_engine/mlir/transforms/op_removal_patterns.td"
 
+class ConstantValue<string val> : AttrConstraint<CPred<"IsConstantValue($_self, " # val # ")">>;
 
-// This relies on implementation details of larq.math.sign. We should make
-// this more general in the future
+// Base quantiser pattern that matches the `tf.where` implementation of `ste_sign`.
+def : Pat<(TF_SelectV2Op:$select_op
+              $cond,
+              (ConstantOp ConstantValue<"1.0f">),
+              (ConstantOp ConstantValue<"-1.0f">)),
```

Comment:
Yes it's possible to do non-symmetric `(a, b)` and I have a plan to implement that, but for now I thought it would be easier to do that in a following PR. For example, you start to have issues relating to padding values, and stuff like that, that need to be thought about carefully.